### PR TITLE
Faulty leader cases clarification

### PR DESCRIPTION
**Pull request by eadventurous**
Signed-off-by: Egor Ivkov <e.o.ivkov@gmail.com>

### CODE REVIEW

File: iroha_2_whitepaper.md [hunk 1]
```diff
@@ -240,10 +240,17 @@ The proxy tail should at this point have received the block from at least one of
 
 **Possible faulty cases related to the leader are:**
 
-- leader ignores all transactions and never creates a block
-
-  - the solution to this is to have other nodes broadcast a transaction across the network and if someone sends a transaction to the leader and it gets ignored, then the leader can be suspected; the suspect message is sent around the network and a new leader is elected if *f*+1 nodes cannot get a reply from the leader for any transaction
-
+- Leader does not send a receipt of receiving transaction(s) from a node in the specified time (*tx receipt time*).
```
Comment (humb1t) [inline iroha_2_whitepaper.md:243]: please, use `peer` term instead of `node`.

File: iroha_2_whitepaper.md [hunk 2]
```diff
@@ -240,10 +240,17 @@ The proxy tail should at this point have received the block from at least one of
...
+  
+  - The peer broadcasts a suspect message with the tx that it tried to send to the leader.
```
Comment (humb1t) [inline iroha_2_whitepaper.md:245]: use fully qualified name - `transaction`
Also you may need to rephrase this bullet - I didn't get the second half of it - who tried to send transaction and was a transaction suspicious or the message?

File: iroha_2_whitepaper.md [hunk 3]
```diff
@@ -240,10 +240,17 @@ The proxy tail should at this point have received the block from at least one of
...
 **Possible faulty cases related to the leader are:**
 
-- leader ignores all transactions and never creates a block
-
-  - the solution to this is to have other nodes broadcast a transaction across the network and if someone sends a transaction to the leader and it gets ignored, then the leader can be suspected; the suspect message is sent around the network and a new leader is elected if *f*+1 nodes cannot get a reply from the leader for any transaction
-
+- Leader does not send a receipt of receiving transaction(s) from a node in the specified time (*tx receipt time*).
+  
+  - The peer broadcasts a suspect message with the tx that it tried to send to the leader. 
+  Other non faulty peers try to reach the leader with this transaction.
```
Comment (humb1t) [inline iroha_2_whitepaper.md:246]: IDK, but isn't it `non-faulty`?

File: iroha_2_whitepaper.md [hunk 4]
```diff
@@ -240,10 +240,17 @@ The proxy tail should at this point have received the block from at least one of
...
-
-  - the solution to this is to have other nodes broadcast a transaction across the network and if someone sends a transaction to the leader and it gets ignored, then the leader can be suspected; the suspect message is sent around the network and a new leader is elected if *f*+1 nodes cannot get a reply from the leader for any transaction
-
+- Leader does not send a receipt of receiving transaction(s) from a node in the specified time (*tx receipt time*).
+  
+  - The peer broadcasts a suspect message with the tx that it tried to send to the leader. 
+  Other non faulty peers try to reach the leader with this transaction. 
+  If *f*+1 peers can not get a reply from the leader with this transaction, then a new leader is elected (*view change*).
+  
+- Leader sends a receipt of receiving transaction(s), but does not create a block in *block time*.
```
Comment (humb1t) [inline iroha_2_whitepaper.md:249]: `received`?

Comment (humb1t) [reply iroha_2_whitepaper.md:249]: and `didn't`

File: iroha_2_whitepaper.md [hunk 5]
```diff
@@ -240,10 +240,17 @@ The proxy tail should at this point have received the block from at least one of
...
+  
+  - The peer that have originally sent the transaction broadcasts a suspect message, using a receipt from the leader as a proof.
```
Comment (humb1t) [inline iroha_2_whitepaper.md:251]: `Original transaction sender peer...`

File: iroha_2_whitepaper.md [hunk 6]
```diff
@@ -240,10 +240,17 @@ The proxy tail should at this point have received the block from at least one of
...
+  Non faulty peers validate the receipt and vote for changing the leader. If *f*+1 votes are gathered, then a new leader is elected (*view change*).
```
Comment (humb1t) [inline iroha_2_whitepaper.md:252]: `-`

File: iroha_2_whitepaper.md (hunk 4 above)
Comment (eadventurous) [reply iroha_2_whitepaper.md:249]: I think the grammar is correct in this sentence. And all the faulty cases here are described in the present tense, so why not follow this style.

File: iroha_2_whitepaper.md (hunk 5 above)
Comment (eadventurous) [reply iroha_2_whitepaper.md]: > `Original transaction sender peer...`

Sounds more complicated, I think it's better to leave it as it is.

File: iroha_2_whitepaper.md [hunk 7]
```diff
@@ -240,10 +240,17 @@ The proxy tail should at this point have received the block from at least one of
...
-
+- Leader does not send a receipt of receiving transaction(s) from a peer in the specified time (*tx receipt time*).
+  
+  - The peer broadcasts a suspect message with the transaction that it tried to send to the leader. 
+  Other non-faulty peers try to reach the leader with this transaction. 
+  If *f*+1 peers can not get a reply from the leader with this transaction, then a new leader is elected (*view change*).
+  
+- Leader sends a receipt of receiving transaction(s), but does not create a block in *block time*.
+  
+  - The peer, that have originally sent the transaction, broadcasts a suspect message, using a receipt from the leader as a proof.
```
Comment (takemiyamakoto) [inline iroha_2_whitepaper.md:251]: "that has originally"...

File: iroha_2_whitepaper.md (hunk 4 above)
Comment (takemiyamakoto) [reply iroha_2_whitepaper.md:249]: Seems okay to me.

**Pull request merged**